internal/services/arckubernetes/arc_kubernetes_flux_configuration_resource.go

```diff
@@ -0,0 +1,1082 @@
+						),
+						ExactlyOneOf: []string{"bucket.0.access_key", "bucket.0.local_auth_ref"},
+					},
+
+					"tls_enabled": {
+						Type:     pluginsdk.TypeBool,
+						Optional: true,
+						Default:  true,
+					},
+
+					"sync_interval_in_seconds": {
+						Type:         pluginsdk.TypeInt,
+						Optional:     true,
+						Default:      600,
+						ValidateFunc: validation.IntBetween(1, 35791394),
+					},
+
+					"timeout_in_seconds": {
+						Type:         pluginsdk.TypeInt,
+						Optional:     true,
+						Default:      600,
+						ValidateFunc: validation.IntBetween(1, 35791394),
+					},
+				},
+			},
+		},
+
+		"git_repository": {
+			Type:         pluginsdk.TypeList,
+			Optional:     true,
+			MaxItems:     1,
+			ExactlyOneOf: []string{"azure_blob", "bucket", "git_repository"},
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"url": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ValidateFunc: validate.KubernetesGitRepositoryUrl(),
+					},
+
+					"reference_type": {
+						Type:     pluginsdk.TypeString,
+						Required: true,
+						ValidateFunc: validation.StringInSlice([]string{
+							FluxGitBranch,
+							FluxGitCommit,
+							FluxGitSemverRange,
+							FluxGitReferenceTag,
+						}, false),
+					},
+
+					"reference_value": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ValidateFunc: validation.StringIsNotEmpty,
+					},
+
+					"https_ca_cert": {
+						Type:         pluginsdk.TypeString,
+						Optional:     true,
+						ValidateFunc: validation.StringIsBase64,
+						Sensitive:    true,
+						RequiredWith: []string{"git_repository.0.https_user"},
+					},
+
+					"https_user": {
+						Type:          pluginsdk.TypeString,
+						Optional:      true,
+						ValidateFunc:  validation.StringIsNotEmpty,
+						RequiredWith:  []string{"git_repository.0.https_key"},
+						ConflictsWith: []string{"git_repository.0.local_auth_ref", "git_repository.0.ssh_private_key", "git_repository.0.ssh_known_hosts"},
+					},
+
+					"https_key": {
+						Type:         pluginsdk.TypeString,
+						Optional:     true,
+						ValidateFunc: validation.StringIsBase64,
+						Sensitive:    true,
+						RequiredWith: []string{"git_repository.0.https_user"},
+					},
+
+					"local_auth_ref": {
+						Type:     pluginsdk.TypeString,
+						Optional: true,
+						ValidateFunc: validation.StringMatch(
+							regexp.MustCompile(`^[a-z\d]([-a-z\d]{0,61}[a-z\d])?$`),
+							"`local_auth_ref` must be between 1 and 63 characters. It can contain only lowercase letters, numbers, and hyphens (-). It must start and end with a lowercase letter or number.",
+						),
+						ConflictsWith: []string{"git_repository.0.https_user", "git_repository.0.ssh_private_key", "git_repository.0.ssh_known_hosts"},
+					},
+
+					"ssh_private_key": {
```

Comment:
If this is base64 encoded we should specify that in the name 
```suggestion
					"ssh_private_key_base64": {
```